### PR TITLE
Store/Publisher invalid token handling improvements

### PR DESCRIPTION
**Pull request by malinthaprasan**
1. When the token is invalid (getting a 401), check the token status and logging out the user if the token is invalid : https://github.com/wso2/product-apim/issues/5387
2. Added token refresh logic from publisher to store
3. Removed token revocation from the UI as the token is already revoked at the server side during logout (added from https://github.com/wso2/carbon-apimgt/pull/6813)
4. Added missing scripts of AM_SYSTEM_APPS: https://github.com/wso2/product-apim/issues/5411
5. Added missing UNIQUE constraints for AM_SYSTEM_APPS: https://github.com/wso2/product-apim/issues/5412
6. Added concurrency protection for multiple login requests at the same time
7. Remove hardcoded "admin" as owner when doing DCR : https://github.com/wso2/product-apim/issues/5413

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/data/APIClient.js [hunk 1]
```diff
@@ -122,6 +123,21 @@ class APIClient {
...
                 APIClient.addETag(data.url, data.headers.etag);
             }
+
+            // If an unauthenticated response is received, we check whether the token is valid by introspecting it.
+            // If it is not valid, we need to clear the stored tokens (in cookies etc) in the browser by redirecting the
+            //   user to logout.
+            if (data.status === 401 && data.obj != null && data.obj.description === 'Unauthenticated request') {
+                const userData = AuthManager.getUserFromToken();
+                userData.then((user) => {
+                    if (user == null) {
```
Comment (mushthaq33) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/data/APIClient.js]: We can modify the if condition as following right?
if (user) {
}

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/source/src/app/data/APIClient.jsx [hunk 2]
```diff
@@ -122,17 +124,83 @@ class APIClient {
...
+
+            // If an unauthenticated response is received, we check whether the token is valid by introspecting it.
+            // If it is not valid, we need to clear the stored tokens (in cookies etc) in the browser by redirecting the
+            //   user to logout.
+            if (data.status === 401 && data.obj != null && data.obj.description === 'Unauthenticated request') {
+                const userData = AuthManager.getUserFromToken();
+                const existingUser = AuthManager.getUser(this.environment.label);
+                if (existingUser) {
+                    userData.then((user) => {
+                        if (user == null) {
```
Comment (mushthaq33) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/source/src/app/data/APIClient.jsx]: If (user) {
}

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/data/APIClient.js (hunk 1 above)
Comment (malinthaprasan) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/data/APIClient.js]: Ahh yes will fix that

Comment (malinthaprasan) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/data/APIClient.js]: Fixed

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/source/src/app/data/APIClient.jsx (hunk 2 above)
Comment (malinthaprasan) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/source/src/app/data/APIClient.jsx]: Fixed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/data/APIClient.js [hunk 3]
```diff
@@ -122,6 +123,21 @@ class APIClient {
...
+                        window.location = Configurations.app.context + Utils.CONST.LOGOUT_CALLBACK;
+                    }
+                }).catch((error) => {
+                    console.error('Error occurred while checking token status. Hence redirecting to login', error);
+                    window.location = Configurations.app.context + Utils.CONST.LOGOUT_CALLBACK;
+                });
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/data/APIClient.js]: If we redirect user to 

```javascript
window.location = Configurations.app.context + Utils.CONST.LOGOUT_CALLBACK;
```

regardless of the promise susses or catch, we could put the above code in `.finally` callback

**Pull request merged**